### PR TITLE
[Merged by Bors] - chore(*): golf using `omega`

### PR DESCRIPTION
**Pull request by urkud**
Found by https://github.com/dwrensha/tryAtEachStep

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
!bench

File: Mathlib/Geometry/Euclidean/MongePoint.lean [hunk 1]
```diff
@@ -468,12 +468,8 @@ theorem eq_orthocenter_of_forall_mem_altitude {t : Triangle ℝ P} {i₁ i₂ :
   rw [orthocenter_eq_mongePoint]
   have ha : ∀ i, i₃ ≠ i → p ∈ t.mongePlane i₃ i := by
     intro i hi
-    have hi₁₂ : i₁ = i ∨ i₂ = i := by
-      clear h₁ h₂
-      decide +revert
-    cases' hi₁₂ with hi₁₂ hi₁₂
-    · exact hi₁₂ ▸ h₂
-    · exact hi₁₂ ▸ h₁
+    obtain rfl | rfl : i₁ = i ∨ i₂ = i := by omega
```
Comment (grunweg) [inline Mathlib/Geometry/Euclidean/MongePoint.lean:471]: This file got a bit slower.

File: Mathlib/Analysis/Convex/Between.lean [hunk 2]
```diff
@@ -519,10 +519,7 @@ theorem sbtw_of_sbtw_of_sbtw_of_mem_affineSpan_pair [NoZeroSMulDivisors R V]
   have h₂₃ : i₂ ≠ i₃ := by
     rintro rfl
     simp at h₁
-  have h3 : ∀ i : Fin 3, i = i₁ ∨ i = i₂ ∨ i = i₃ := by
-    clear h₁ h₂ h₁' h₂'
-    -- Porting note: Originally `decide!`
-    revert i₁ i₂ i₃; decide
+  have h3 : ∀ i : Fin 3, i = i₁ ∨ i = i₂ ∨ i = i₃ := by omega
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Between.lean:522]: This also got a bit slower.

File: Mathlib/NumberTheory/PythagoreanTriples.lean [hunk 3]
```diff
@@ -194,9 +194,7 @@ theorem normalize : PythagoreanTriple (x / Int.gcd x y) (y / Int.gcd x y) (z / I
 theorem isClassified_of_isPrimitiveClassified (hp : h.IsPrimitiveClassified) : h.IsClassified := by
   obtain ⟨m, n, H⟩ := hp
   use 1, m, n
```
Comment (grunweg) [inline Mathlib/NumberTheory/PythagoreanTriples.lean:196]: As did this file (not sure which line).

File: Mathlib/NumberTheory/Zsqrtd/QuadraticReciprocity.lean [hunk 4]
```diff
@@ -39,12 +39,7 @@ theorem mod_four_eq_three_of_nat_prime_of_prime (p : ℕ) [hp : Fact p.Prime]
           exact absurd this (by decide)))
     fun hp1 =>
     by_contradiction fun hp3 : p % 4 ≠ 3 => by
-      have hp41 : p % 4 = 1 := by
-        rw [← Nat.mod_mul_left_mod p 2 2, show 2 * 2 = 4 from rfl] at hp1
-        have := Nat.mod_lt p (show 0 < 4 by decide)
-        revert this hp3 hp1
-        generalize p % 4 = m
-        intros; interval_cases m <;> simp_all -- Porting note (https://github.com/leanprover-community/mathlib4/issues/11043): was `decide!`
+      have hp41 : p % 4 = 1 := by omega
```
Comment (grunweg) [inline Mathlib/NumberTheory/Zsqrtd/QuadraticReciprocity.lean:42]: This file got about 5% slower (but this is a nice golf, so perhaps worth it).

File: Mathlib/NumberTheory/LegendreSymbol/QuadraticChar/Basic.lean [hunk 5]
```diff
@@ -290,8 +290,6 @@ theorem FiniteField.isSquare_neg_one_iff : IsSquare (-1 : F) ↔ Fintype.card F
   · have h₁ := FiniteField.odd_card_of_char_ne_two hF
     rw [← quadraticChar_one_iff_isSquare (neg_ne_zero.mpr (one_ne_zero' F)),
       quadraticChar_neg_one hF, χ₄_nat_eq_if_mod_four, h₁]
-    simp only [Nat.one_ne_zero, if_false, ite_eq_left_iff, (by omega : (-1 : ℤ) ≠ 1), imp_false,
-      not_not, Ne, reduceCtorEq]
-    exact ⟨fun h ↦ ne_of_eq_of_ne h (by omega), (Nat.odd_mod_four_iff.mp h₁).resolve_right⟩
+    omega
```
Comment (grunweg) [inline Mathlib/NumberTheory/LegendreSymbol/QuadraticChar/Basic.lean:293]: This file also got a bit slower.

File: Mathlib/RingTheory/Polynomial/Bernstein.lean [hunk 6]
```diff
@@ -170,9 +170,7 @@ theorem iterate_derivative_at_0 (n ν : ℕ) :
         eval_natCast, Function.comp_apply, Function.iterate_succ, ascPochhammer_succ_left]
       obtain rfl | h'' := ν.eq_zero_or_pos
       · simp
-      · have : n - 1 - (ν - 1) = n - ν := by
-          rw [gt_iff_lt, ← Nat.succ_le_iff] at h''
-          rw [← tsub_add_eq_tsub_tsub, add_comm, tsub_add_cancel_of_le h'']
+      · have : n - 1 - (ν - 1) = n - ν := by omega
```
Comment (grunweg) [inline Mathlib/RingTheory/Polynomial/Bernstein.lean:173]: Also very slightly slower - but the automation is probably worth it.

**Review by grunweg — COMMENTED** [review]
Let me label a few places that got slightly slower (found by sorting the benchmarking output by the "absolute change" column). Probably, most of these are still worth it - but this should be a conscious decision.

**Comment by grunweg** [comment]
Taking a step back: thanks for proposing these golfs! They are much nicer to read, and mostly quite a bit faster.

**Review by alreadydone — APPROVED** [review]
The total build instructions only increases by 0.006%, which amounts to 1038*.006%~0.06 seconds wall-clock time, so I wouldn't worry about the slowdown. Thanks!
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**